### PR TITLE
Jakarta SOAP 2.0

### PR DESCRIPTION
**Pull request by lukasj**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of wombat-spec-x.y.pdf
- [x] Specification HTML in the form of wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a Release Review, a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification.
The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/xml/soap/jakarta.xml.soap-api/2.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-soap-tck-2.0.0.zip
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/saaj-api/issues/84
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

### CODE REVIEW

**Comment by kwsutter** [comment]
@karianna, can you help mentor this PR across the finish line for Jakarta EE 9?  I'll post the Spec Review Checklist to get you started.  Reach out if you have questions.  Thanks!

**Comment by kwsutter** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by kwsutter** [comment]
Just to allow @lukasj to make more progress, I will start reviewing this PR and marking some items off on the checklist... But, we need @karianna to perform the more thorough mentoring and reviewing.  Thanks!

**Comment by lukasj** [comment]
> The SHA value for the TCK doesn't seem to match the SHA value in the CR.

the staged one has some issues in docs which are fixed in nightly used for CR. It should get in sync as soon as that nightly gets to the right area

the rest should be addresed already

**Comment by kwsutter** [comment]
Thanks, @lukasj.  Marked off a few more items in the Checklist.  Once we get a final TCK and SHA value, then we are looking pretty good!

The Certification Request needs the `certification` label (I didn't have the permissions to do this for you).

**Comment by karianna** [comment]
Hi all - apologies - these GitHub notifications simply never came through to me :-(.  I'll start looking this evening.

File: soap-attachments/2.0/apidocs/jquery/jquery-ui.css
Comment (karianna) [inline soap-attachments/2.0/apidocs/jquery/jquery-ui.css:1]: Was there a reason we dropped to a lower version?

Comment (lukasj) [reply soap-attachments/2.0/apidocs/jquery/jquery-ui.css:1]: Different JDK on VMs

**Comment by karianna** [comment]
@lukasj - I might be looking in the wrong place but https://download.eclipse.org/jakartaee/soap-attachments/2.0/jakarta-soap-tck-2.0.0.zip wasn't available?

**Comment by lukasj** [comment]
> @lukasj - I might be looking in the wrong place but https://download.eclipse.org/jakartaee/soap-attachments/2.0/jakarta-soap-tck-2.0.0.zip wasn't available?

because the tck has not been released yet. Once it will be released _and_ this PR merged, the link will start working. @kwsutter can give you more details on why it works this way.

**Comment by kwsutter** [comment]
> > @lukasj - I might be looking in the wrong place but https://download.eclipse.org/jakartaee/soap-attachments/2.0/jakarta-soap-tck-2.0.0.zip wasn't available?
> 
> because the tck has not been released yet. Once it will be released _and_ this PR merged, the link will start working. @kwsutter can give you more details on why it works this way.

The PR is prepping for the final specifications page by updating the _index.md file.  This file gets processed by Hugo and the page is rendered similar to this:  https://deploy-preview-231--jakartaee-specifications.netlify.app/specifications/soap-attachments/2.0/

This page needs to have the "final" url for the TCK.  The final repo for the TCK would be here:  https://download.eclipse.org/jakartaee/soap-attachments/2.0/jakarta-soap-tck-2.0.0.zip 

Since part of the review process is to actually review the contents of the TCK, we also ask for the staged version of the TCK in the PR checklist.
 - [ ] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-soap-tck-2.0.0.zip

(Technically, this staged version of the TCK should be in the `promoted` directory, but that process is still being ironed out...)

Hope this helps!

**Comment by lukasj** [comment]
fixed the usage of SAAJ acronym in the spec document

**Comment by m0mus** [comment]
@karianna @kwsutter please start the ballot if everything you think that this PR is ready.

**Comment by kwsutter** [comment]
> @karianna @kwsutter please start the ballot if everything you think that this PR is ready.

I will start the ballot to keep the train running...

**Review by karianna — APPROVED** [review]
Apologies - I was away on leave - this LGTM

**Comment by kwsutter** [comment]
* On ballot completion, the specification committee mentor:
 - [x] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>


 ----
 - [ ] promotes api staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [ ] go through the merged jakarta.ee specification website page to verify all the links are valid.
 - [ ] approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email to tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] merge any final release branch as appropriate for the branch management for the project.

File: soap-attachments/2.0/_index.md
Comment (kwsutter) [inline soap-attachments/2.0/_index.md:3]: @lukasj Just a comment... The ballot is now closed for this Specification.  So, when this PR gets merged, this date field will determine when to make the web page go live.  You might want to consider changing this to 2020-09-16 to allow for immediate consumption once the merge is complete.  If you don't see this comment before I'm ready to merge, then I'll make the update when I do the ballot update.  Thanks.

**Review by kwsutter — COMMENTED** [review]
Thanks.  One way or the other, we'll want to modify this date field...

File: soap-attachments/2.0/_index.md
Comment (keilw) [reply soap-attachments/2.0/_index.md:3]: What's the policy here, because both Authorization, Mail and others did not change the date in the top to September, should we do that?

Comment (lukasj) [reply soap-attachments/2.0/_index.md:3]: @kwsutter my assumption was that the spec is about to go live once the release of the api and impl is approved by pmc etc which is going to need another few weeks (actually the api/impl cannot be release before Oct 7, since release review is to conclude on Oct 7 unless I'm missing something) If you don't mind having broken links for that long, feel free to change the date

Comment (kwsutter) [reply soap-attachments/2.0/_index.md:3]: I realize the Release Reviews are necessary, but we still don't have that process running smoothly yet.  As Werner pointed out, we already had a couple of Spec PRs get approved without a formal Release Review.  I will go ahead and merge this as-is, but I will probably change this when I submit the ballot update.  In my mind, if the ballot is approved, then the Release Review is a formality that will eventually even out.  Eventual consistency... :-)

Comment (lukasj) [reply soap-attachments/2.0/_index.md:3]: @kwsutter https://www.eclipse.org/projects/handbook/#specification-project-reviews is usually my source. Whatever gets decided should be reflected in the handbook too

**Pull request merged**
